src/renderer/components/dock/upgrade-chart/create-upgrade-chart-tab.injectable.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import upgradeChartTabStoreInjectable from "./store.injectable";
+import dockStoreInjectable from "../dock/store.injectable";
+import type { HelmRelease } from "../../../../common/k8s-api/endpoints/helm-releases.api";
+import { DockStore, DockTabCreateSpecific, TabId, TabKind } from "../dock/store";
+import type { UpgradeChartTabStore } from "./store";
+import { bind } from "../../../utils";
+import { runInAction } from "mobx";
+
+interface Dependencies {
+  upgradeChartStore: UpgradeChartTabStore;
+  dockStore: DockStore
+}
+
+function createUpgradeChartTab({ upgradeChartStore, dockStore }: Dependencies, release: HelmRelease, tabParams: DockTabCreateSpecific = {}): TabId {
+  const tabId = upgradeChartStore.getTabIdByRelease(release.getName());
+
+  if (tabId) {
+    dockStore.open();
+    dockStore.selectTab(tabId);
+
+    return tabId;
+  }
+
+  return runInAction(() => {
+    const tab = dockStore.createTab(
+      {
+        title: `Helm Upgrade: ${release.getName()}`,
+        ...tabParams,
+        kind: TabKind.UPGRADE_CHART,
+      },
+      false,
+    );
+
+    upgradeChartStore.setData(tab.id, {
+      releaseName: release.getName(),
+      releaseNamespace: release.getNs(),
+    });
+
+    return tab.id;
+  });
+}
+
+const createUpgradeChartTabInjectable = getInjectable({
+  instantiate: (di) => bind(createUpgradeChartTab, null, {
```

Comment:
This bind pattern is just complexity which shouldn't be added. Configuration (e.g. dependencies) shouldn't be mixed with arguments. Arrow function would be much simpler.